posthog/hogql/database/database.py

```diff
@@ -491,7 +493,7 @@ def serialize_database(
     warehouse_table_names = context.database.get_warehouse_tables()
     warehouse_tables = (
         list(
-            DataWarehouseTable.objects.prefetch_related("external_data_source")
+            DataWarehouseTable.objects.select_related("credential", "external_data_source")
```

Comment:
prefetch_related is in-memory and is for many-to-many or reverse foreign key relationships

so if you were selecting data warehouse table from an external data source then i think it would error if you tried to use select_related and you'd have to use prefetch

but here we're navigating a foreign key in its "normal" direction so can select related